xeol/search/purl.go

```diff
@@ -116,10 +116,17 @@ func returnMatchingCycle(version string, cycles []eol.Cycle) (eol.Cycle, error)
 			return c, nil
 		}
 
-		// match on major, minor, or patch
+		// if it's a constraint, let's use that first
+		cx, err := semver.NewConstraint(c.ReleaseCycle)
```

Comment:
This may require support for the `~X.Y.Z` syntax common in the new data set, which may need to be replaced with something more compatible by the `semver` library.

```suggestion
		tildeRe := regexp.MustCompile(`(^|\s+)~(\d+)`)
                releaseCycle := tildeRe.ReplaceAllString(c.ReleaseCycle, "$1>=$2")
                cx, err := semver.NewConstraint(releaseCycle)
```

Syntax needs to be checked but this roughly should cover the cases of new data format